src/components/SearchInput/SearchInput.tsx

```diff
@@ -0,0 +1,121 @@
+import {
+  AriaAttributes,
+  ChangeEvent,
+  FocusEvent,
+  MouseEvent,
+  ReactNode,
+  useEffect,
+  useRef,
+  useState,
+} from "react"
+import styles from "./SearchInput.module.css"
+
+interface SearchInputProps extends Pick<AriaAttributes, "aria-label"> {
```

Comment:
Have you considered extending the native `HTMLInputElement` element here? This way we can simplify the interface and get rid of placeholder, onBlur, onFocus, etc.

Reply:
I didn't want to do that as I'm trying to avoid the use of onChange. It honestly felt like it bloated it a bit by passing through all of the input props. I just felt like having the onChange loses some of the benefits gained from just using the onSearch callback.

Reply:
@luiz-chagas @roblafeve @albatrocity Feel free to weigh in on this as well. I can be persuaded.

Reply:
If onChange is the only issue we could omit it from the interface: 
`Omit<HTMLProps<HTMLInputElement>, "onChange">`

Reply:
Fair enough